### PR TITLE
docs(Form): add nest table demo

### PR DESCRIPTION
**Pull request by OneMoreJack**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
Sometimes, we need to handle `Array` data structure in form and `Table` is built for displaying this kind of data. When I was new to ant-design, It was hard for me to do this. I hope this demo would help someone who might be confused with how to nest table in a form.
 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add demo about nesting `Table` in `Form`   |
| 🇨🇳 Chinese |  新增在表单中嵌入表格的示例         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
ci 挂了，请检查。

**Comment by JackLiR8** [comment]
有几个问题请教下:
Snipaste_2022-12-26_20-43-36

**Comment by JackLiR8** [comment]
有几个问题请教下：
1. 之前 CI 没通过是因为 snapshot 没更新，更新文档的 snapshot 是运行 `test:update` 这个脚本吗？
2. 现在打开项目会自动生成 `components/version/token.tsx`, 这个文件是不是应该添加到 `.gitignore`, 我看仓库代码里 version 目录下只有一个 `index.tsx`

**Comment by afc163** [comment]
1. 是的。
2. 可以，不过我好像没遇到这个问题。

File: components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 1]
```diff
@@ -3279,6 +3279,7 @@ exports[`renders ./components/calendar/demo/customize-header.tsx extend context
                 <div>
                   <div
                     class="rc-virtual-list"
+                    innerprops="[object Object]"
```
Comment (afc163) [inline components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap:3282]: 不应该有这个。

Comment (JackLiR8) [reply components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap:3282]: 请问为什么我 `test:update` 后会有这个 `innerprops` 呢？😅

还有, 我看还有另外一种错误：
 `Snapshot` 和 `Received` 中 div 的 style 不匹配，前者的 sytle 少了 `border:0;padding:0` 这两个样式

```
FAIL components/transfer/__tests__/demo.test.ts
  ● renders ./components/transfer/demo/tree-transfer.tsx correctly

    expect(received).toMatchSnapshot()

    Snapshot name: `renders ./components/transfer/demo/tree-transfer.tsx correctly 1`

    - Snapshot  - 1
    + Received  + 1

    @@ -41,11 +41,11 @@
                    />
                  </div>
                  <div
                    aria-hidden="true"
                    class="ant-tree-treenode"
    -               style="position:absolute;pointer-events:none;visibility:hidden;height:0;overflow:hidden"
    +               style="position:absolute;pointer-events:none;visibility:hidden;height:0;overflow:hidden;border:0;padding:0"
                  >
                    <div
                      class="ant-tree-indent"
                    >
                      <div

      116 |           type: 'demo',
      117 |           html,
    > 118 |         }).toMatchSnapshot();
          |            ^
      119 |
      120 |         errSpy.mockRestore();
      121 |       },

      at Object.toMatchSnapshot (tests/shared/demoTest.tsx:118:12)

```

这种错误该如何修改呢，手动修改 snapshots 应该不可取吧，毕竟下次 `test:update` 后这两个样式又没了

Comment (afc163) [reply components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap:3282]: 你需要更新本地 node_modules 到最新，然后再跑 test:update

Comment (JackLiR8) [reply components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap]: 感谢👏，snapshots 应该可以了。但 tooltip 组件的覆盖率降低了😅

Comment (yoyo837) [reply components/calendar/__tests__/__snapshots__/demo-extend.test.ts.snap]: 多等会儿, 可能没跑完

File: components/form/demo/nest-table-in-form.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            return (
+              <ConfigProvider
+                renderEmpty={() => (
+                  <Empty description={false}>
+                    <Button type="primary" ghost onClick={() => operation.add()}>
+                      Add User
+                    </Button>
+                  </Empty>
+                )}
+              >
```
Comment (afc163) [inline components/form/demo/nest-table-in-form.tsx:117]: 这里为啥要用 ConfigProvider ？直接用 Table 的 emptyText 可行么？

File: components/form/demo/nest-table-in-form.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      title: 'Age',
+      dataIndex: 'age',
+      render(_, { field }) {
+        return (
+          <Form.Item name={[field.name, 'age']} style={formItemStyle}>
+            <InputNumber style={{ width: '100%' }} />
+          </Form.Item>
+        );
+      },
+    },
```
Comment (afc163) [inline components/form/demo/nest-table-in-form.tsx:62]: return 多余了，直接用 =>

File: components/form/demo/nest-table-in-form.tsx (hunk 2 above)
Comment (JackLiR8) [reply components/form/demo/nest-table-in-form.tsx:117]: 有这个属性吗？不好意思我在文档里没搜到，如果有 emptyText 且支持 ReactNode 的话应该可以，我试下

Comment (afc163) [reply components/form/demo/nest-table-in-form.tsx:117]: 应该是 `locale={{ emptyText: ... }}`

Comment (JackLiR8) [reply components/form/demo/nest-table-in-form.tsx]: 👌

**Comment by afc163** [comment]
冲突了

**Comment by afc163** [comment]
ping

**Comment by li-jia-nan** [comment]
Hi @OneMoreJack 
Thank you for your contribution!
We noticed that this PR hasn't been updated in a while. To keep the project active and maintainable, we'll be closing this PR for now.
If you'd like to proceed with these changes, please feel free to open a new PR.
Thanks again for your contribution!

**Pull request closed without merging**